src/cpp/server/server_cc.cc

```diff
@@ -47,6 +47,12 @@
 namespace grpc {
 namespace {
 
+// The default value for maximum number of threads that can be created in the
+// sync server. This value of 500 is empirically chosen. To increase the max
+// number of threads in a sync server, pass a custom ResourceQuota object (with
+// the desired number of max-threads set) to the server builder
+#define DEFAULT_MAX_SYNC_SERVER_THREADS 500
```

Comment:
I'm wondering if we do want to make this some function of # cores using gpr_cpu_num_cores() and then default to some conservative limit if that information isn't available. I can imagine users running their application on a different machine with way less # cores and then being confused on why performance tanks.